tanjun/checks.py

```diff
@@ -509,6 +511,43 @@ async def __call__(
         return self._handle_result((permissions & self._permissions) == self._permissions)
 
 
+class HasAnyRoleCheck(_Check):
+    __slots__ = ("required_roles", "ids_only")
+
+    def __init__(
+        self,
+        roles: collections.Sequence[typing.Union[hikari.SnowflakeishOr[hikari.Role], str]] = [],
+        *,
+        error_message: typing.Optional[str] = "You do not have the required roles to use this command!",
+        halt_execution: bool = True,
+    ) -> None:
+        super().__init__(error_message, halt_execution)
+        self.required_roles = roles
+        self.ids_only = all(isinstance(role, int) for role in self.required_roles)
+
+    async def __call__(self, ctx: tanjun_abc.Context, /) -> bool:
+        if not ctx.member:
+            return self._handle_result(False)
+
+        if not self.ids_only:
+            guild_roles = ctx.cache.get_roles_view_for_guild(ctx.member.guild_id) if ctx.cache else None
+            if not guild_roles:
+                guild_roles = await ctx.rest.fetch_roles(ctx.member.guild_id)
+                member_roles = [role for role in guild_roles if role.id in ctx.member.role_ids]
+            else:
+                member_roles = [guild_roles.get(role) for role in ctx.member.role_ids]
+        else:
+            member_roles = ctx.member.role_ids
+
+        return self._handle_result(any(map(self._check_roles, member_roles)))
+
+    def _check_roles(self, member_role: typing.Union[int, hikari.Role]) -> bool:
+        if isinstance(member_role, int):
```

Comment:
rather than instance check here could the different checks be performed in the separate parts of the `if not self._ids_only else` statement to avoid the instance checks all together (if type checking doesn't quite like this you can just cast but i don't think it should be too strict for equality checks)